test/test_jit.py

```diff
@@ -4323,6 +4323,38 @@ def func(a, b):
         inputs = self._make_scalar_vars([-1, 1], torch.int64)
         self.checkScript(func, inputs, optimize=True)
 
+    def test_if_is_none_dispatch(self):
+        class Test(torch.jit.ScriptModule):
+            __constants__ = ['b']
+
+            def __init__(self, b=None):
+                super(Test, self).__init__()
+                self.b = b
+
+            @torch.jit.script_method
+            def forward(self, input, opt=None):
+                # type: (Tensor, Optional[Tensor]) -> Tensor
+                x = input
+                if self.b is not None:
```

Comment:
These tests are missing coverage: rhs is always going to be classified as ALWAYS. You can put these additional tests in a follow up PR.

Reply:
Sounds good I will do a follow up PR after the release :)